src/Cello/simulation_Simulation.cpp

```diff
@@ -948,24 +969,31 @@ void Simulation::r_monitor_performance(CkReductionMsg * msg)
 		   num_particles);
 
   // compute total blocks and leaf blocks
-  int num_total_blocks = 0;
+  long long num_total_blocks = 0;
   long long num_leaf_blocks = counters_reduce[m];;
-  int rank = cello::rank();
-  int num_child_blocks = (rank == 1) ? 2 : ( (rank == 2) ? 4 : 8);
-  for (int i=0; i<=hierarchy_->max_level(); i++) {
-    long long num_blocks_level = counters_reduce[m++];
-    monitor()->print("performance","simulation num-blocks-%d %d",
+  for (int i=hierarchy_->min_level(); i<=hierarchy_->max_level(); i++) {
+    const long long num_blocks_level = counters_reduce[m++]; // NL
+    monitor()->print("performance","simulation num-blocks-level %d %ld",
 		     i,num_blocks_level);
     num_total_blocks += num_blocks_level;
-    if (i>0)
-      num_leaf_blocks += (num_blocks_level - num_blocks_level/num_child_blocks);
+    // compute leaf blocks given number of blocks per level
+    if (i>0) num_leaf_blocks +=
+	(num_blocks_level - num_blocks_level/cello::num_children());
```

Comment:
if something like num_blocks_level is an integer and cello::num_children() is also an integer, will this produce the desired behavior?

Reply:
@bwoshea In general no, but here yes: since all blocks refine fully into num_children() sub-blocks, the number of blocks in a refined level (i > 0) will always be evenly divisible by num_children().